### ISSUE TITLE
Is there a way to do partial_run with queues?

### ISSUE THREAD

**Issue by derekhh** [tensorflow/tensorflow#8567]
I'm reading a bunch of image files using the FIFO queue provided by `string_input_producer`. Some of these files have bad format and cannot be decoded properly. Therefore I would want to print the names of these files when the related functions from `tf.image` throw `tf.errors.InvalidArgumentError` exceptions.

My graph definition looks like this:

    ...
    filename_queue = tf.train.string_input_producer(files, num_epochs=1, shuffle=True)
    image_reader = tf.WholeFileReader()
    key, image_file = image_reader.read(filename_queue)
    is_png = tf.py_func(has_png_ext, [key], [tf.bool])
    image = tf.cond(pred=is_png[0],
                    fn1=lambda: tf.image.decode_png(image_file),
                    fn2=lambda: tf.image.decode_jpeg(image_file))
    ...

And then I'm fetching the output like this:

    ...
    with tf.Session() as sess:
        sess.run(init_op)
        coord = tf.train.Coordinator()
        threads = tf.train.start_queue_runners(coord=coord)
        try:
            while not coord.should_stop():
                try:
                    nodes = sess.run([key, image])
                    nodes_results.append(nodes)
                    num_files += 1
                except tf.errors.InvalidArgumentError:
                    num_failed += 1
                    num_failed_dir += 1
    ...

Initially I've tried to print the value of `key` in the `except` block but this won't work. Then I've tried a bunch of ideas and none of them seems to be working. I've been trying to see if [partial_run](https://www.tensorflow.org/api_docs/python/tf/Session#partial_run) could help but it takes a `feed_dict` input instead of a native queue input.

So in short, my question is, is there a way to make partial_run work while using native queue inputs? Or is there a better way to solve my problem - output the file names when `tf.image.decode_*` throw exceptions?

**Comment by derekhh** [comment]
Another idea I've tried is based on this [stackoverflow answer](stackoverflow.com/a/37584169) where the OP suggests using `tf.placeholder_with_default` and set its default input as the output from the FIFO queue.

As a result I've tried something like this:

    ...
    input_from_filename_queue = image_reader.read(filename_queue)
    input = tf.placeholder_with_default(input=input_from_filename_queue, shape=(2, ))
    ...
    h = sess.partial_run_setup(fetches=[file_name, image], feeds=input)
    file_name_tensor = sess.partial_run(handle=h, fetches=file_name, feed_dict=None)
    image_tensor = sess.partial_run(handle=h, fetches=image, feed_dict=None)
    ...

I've tried different variations of things to feed into the `feed_dict` in the above snippet but they all failed. I'm wondering whether this approach could be modified into a working solution.

PS: I'm not exactly sure whether this fits into a *new feature request*. I've tried asking the same question on [stackoverflow](http://stackoverflow.com/questions/42757785/how-should-i-output-partial-information-in-tensorflow-when-exceptions-are-thrown) without receiving any answer so I want to try and see if this question fits here.

**Comment by skye** [comment]
@suharshs can you comment on whether partial run can support this?

**Comment by skye** [comment]
Also @caisq could tfdbg help here?

**Comment by suharshs** [comment]
Does the InvalidArgumentError not have the name of the file that is invalid?

**Comment by caisq** [comment]
@derekhh @skye 
Assuming the `sess.run([key, image])` does not run in a distributed setting, tfdbg (TensorFlow Debugger) should be able to help here. (tfdbg supports only `DirectSessions` right now; support for distributed session is in the works.)

Modifying @derekhh 's code example a little, we have:
```
from tensorflow.python import debug as tf_debug  # tfdbg-required import

with tf.Session() as sess:
    # Activate tfdbg CLI!
    sess = tf_debug.LocalCLIDebugWrapperSession(sess)

    coord = tf.train.Coordinator()
    threads = tf.train.start_queue_runners(coord=coord)

    num_files = 0
    num_failed = 0
    while not coord.should_stop():
      try:
        key_val, image_val = sess.run([key, image])
        print("key_val = %r" % key_val)
        # print("image_val = %r" % image_val)
        num_files += 1
      except tf.errors.InvalidArgumentError:
        num_failed += 1
```

Then, when you run your program, you will automatically drop to the tfdbg CLI. Keep entering `tfdbg> run` until you reach the error screen that looks like the
![debug_string_input_producer_1](https://cloud.githubusercontent.com/assets/16824702/24278778/429c7384-101c-11e7-9690-712de132c25b.png)

(If you are tired of typing run many times to encounter the error, just do something like `tfdbg> run -t 10000`)

Click the `li -r cond/DecodeJpeg` link with your mouse (or type it in, if you like typing) and you'll see a screen like the following:
![debug_string_input_producer_2](https://cloud.githubusercontent.com/assets/16824702/24278808/73d4159c-101c-11e7-8e4f-e76e5493988c.png)

The first input, namely `cond/DecodeJpeg/Switch`, corresponds to the tf.cond node. The node `ReaderReadV2` is the input to the `PyFunc`. To see the value of the first output (i.e., the key) of it, do
```
tfdbg> pt ReaderReadV2:0 and that should give you the value of the file name/path string.
```
![debug_string_input_producer_3](https://cloud.githubusercontent.com/assets/16824702/24278907/2da6c064-101d-11e7-939a-b0258f753b4b.png)

**Comment by girving** [comment]
Closing since complex use of queues are deprecated in favor of @mrry's `tf.contrib.data`.
